File: benchmarks/mandelbrot.js

function mandelbrot() {
	var h = 150.0;
	var y = 0.0;
	var Z = 0.0;
	var z = 0.0;
	var T = 0.0;
	var t = 0.0;
	var C = 0.0;
	var c = 0.0;
	var U = 0.0;
	var V = 0.0;
	var K = 1.5;
	var k = 1.0;

	while (y < 150) {
		y += 1;
		var x = 0.0;
		while (x < 150){
			x += 1;
			Z = 0.0; z=0.0; T=0.0; t = 0.0;
			U = x*2;
			U /= h;
			V = y*2;
			V /= h;
			C = U - K;
			c = V - k;

			var i = 0;
			while (i < 50) {
				i += 1;
				if (T+t <= 4){
					z = Z*z;
					z *= 2;
					z += c;
					Z = T - t;
					Z += C;
					T = Z * Z;
					t = z * z;
				}
			}
			if (T+t <= 4){
				process.stdout.write('*');
			} else {
				process.stdout.write('·');
			}
		}
	}
}

mandelbrot();

File: benchmarks/mandelbrot.lua
h = 150
Z = 0.0
z = 0.0
T = 0.0
t = 0.0
C = 0.0
c = 0.0
U = 0.0
V = 0.0
K = 1.5
k = 1.0

function mandelbrot()
	y = 0
	while (y < 150)
	do
		y = y + 1
		x = 0
		while (x < 150)
		do
			x = x+1
			Z = 0.0
			z = 0.0
			T = 0.0
			t = 0.0
			U = x*2
			U = U / h
			V = y*2
			V = V / h
			C = U - K
			c = V - k

			i = 0
			while (i < 50)
			do
				i = i+1
				if T+t <= 4 then
					z = Z*z
					z = z*2
					z = z+c
					Z = T - t
					Z = Z+C
					T = Z * Z
					t = z * z
				end
			end

			if T+t <= 4 then
				io.write('*')
			else
				io.write('·')
			end
		end
	end
end
mandelbrot()
